### PR TITLE
feat(RHTAPREL-392): add check-data-keys task

### PR DESCRIPTION
**Pull request by johnbieren**
Add a new tekton task that can check the data.json file in the workspace to ensure that all keys are present in order to proceed to the tasks that use the data.

### CODE REVIEW

File: tasks/check-data-keys/check-data-keys.yaml [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+        for SYSTEM in "$@" ; do
+            if [ $(jq -r ".$SYSTEM" <<< $KEYS_JSON) == null ] ; then
+                echo "Unsupported system value: $SYSTEM"
+                exit 1
+            fi
+
+            echo "Checking all required keys exist for: $SYSTEM"
+            for KEY in $(jq -r ".$SYSTEM[]" <<< $KEYS_JSON) ; do
+                if [[ $(jq ".$SYSTEM.$KEY" "$(workspaces.data.path)/$(params.dataPath)") == "null" ]] ; then
+                    echo "Missing required $SYSTEM key: $KEY"
```
Comment (scoheb) [inline tasks/check-data-keys/check-data-keys.yaml:63]: should you not try to check all keys before failing?

**Review by scoheb — CHANGES_REQUESTED** [review]

File: tasks/check-data-keys/check-data-keys.yaml (hunk 1 above)
Comment (johnbieren) [reply tasks/check-data-keys/check-data-keys.yaml:63]: Why? If any required key is missing, we can't proceed. I don't think it matters if we are missing 1 required key or 10

Comment (scoheb) [reply tasks/check-data-keys/check-data-keys.yaml:63]: I don't think we want to force users to have to rerun the release pipeline 10 times to find the 10 missing keys

Comment (johnbieren) [reply tasks/check-data-keys/check-data-keys.yaml:65]: Changed to wait til all systems+keys are checked before exiting

Comment (scoheb) [reply tasks/check-data-keys/check-data-keys.yaml:65]: thanks

**Review by scoheb — APPROVED** [review]

**Pull request merged**